.github/workflows/pr-check

```diff
@@ -0,0 +1,178 @@
+        sequence of DOM siblings between any two consecutive headings.
+
+        For example, given:
+
+            ```md
+            # First Section
+
+            Hello World
+
+            # Second Section
+            ```
+
+        we would get:
+
+            >>> pr.section_by_title('First Section')
+            ['\n', <p>Hello World</p>]
+        """
+        headings = [
+            heading
+            for heading in self.body.select('h1,h2,h3,h4,h5,h6')
+            if heading.text.lower() == title.lower()
+        ]
+
+        if len(headings) != 1:
+            raise UnexpectedPRFormat("PR body should have a section titled '{}'.".format(title))
+
+        heading = headings[0]
+
+        elems = []
+        for elem in heading.next_siblings:
+            if not elem.name:
+                # text nodes
+                elems.append(elem)
+                continue
+
+            if re.match(r'h\d', elem.name):
+                # next heading
+                break
+            elems.append(elem)
+
+        return elems
+
+    def check_ids_in_title(self) -> Optional[str]:
+        """Checks that PR title does not include references to issues."""
+        match = re.search(r'#\d+', self.pr.title)
+        if match:
+            return "PR titles shouldn't contain issue IDs, found: '{}'.".format(match.group())
+
+        return None
+
+    def check_change_summary(self) -> Optional[str]:
+        """Checks that a Change Summary section exists."""
+        try:
+            self.section_by_title('Change Summary')
+        except UnexpectedPRFormat as e:
+            return str(e)
+
+        return None
+
+    def check_related_issue_ref(self) -> Optional[str]:
+        """If there are #N references in the 'Related issue number' section, at
+        least one uses a valid Github linking verb, eg fixes."""
+        # cf https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
+        valid_issue_ref_verbs = [
+            'close', 'closes', 'closed',
+            'fix', 'fixes', 'fixed',
+            'resolve', 'resolves', 'resolved',
+        ]
+        try:
+            section = self.section_by_title('Related issue number')
+        except UnexpectedPRFormat as e:
+            return str(e)
+
+        text = ' '.join(e.text for e in section)
+        if not re.search(r'#\d+', text):
+            # ok to not reference anything
+            return None
+
+        # note: we don't actually check whether a #N reference is to an issue;
+        # it might be to another PR or to a discussion.
+        match = re.search(r'(\w+)\s+#(\d+)', text)
+        if not match:
+            return "Issue refs should use valid linking verbs like 'fixes #N'."
+
+        verb, issue_id = match.groups()
+        if verb not in valid_issue_ref_verbs:
+            return "Issue refs should use valid linking verbs like 'fixes #{}' not '{}'.".format(
```

Comment:
personal preference, but i would always use f-strings, not `format()` these days.